### PR TITLE
Special characters in static inputs should not be escaped

### LINKED ISSUE #15447: Special characters in static inputs should not be escaped
**Describe the bug**

As visible in [this comment](https://github.com/camunda/feel-scala/pull/750#pullrequestreview-1731283017) and in the description of [this issue](https://github.com/camunda/zeebe/issues/9859) special characters in static input are escaped. 

**To Reproduce**

Follow the step in the issue linked above

**Expected behavior**

Characters should not be escaped

**Log/Stacktrace**

none

### PR DESCRIPTION
**Pull request by nicpuppa**
## Description

This PR fix the behaviour of static expression in Zeebe, ~~Using the built-in `translateEscapes()` API allow Zeebe to not escape special characters also in a Static Inputs~~ by removing all the escaping function 

## Related issues

closes #15447 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by saig0 — CHANGES_REQUESTED** [review]
@nicpuppa thank you for fixing the issue. :rocket: 

There are test failures. Please have a look. :cookie:

**Comment by nicpuppa** [comment]
Hey @saig0, I tried to have a deeper look into the other cases that you suggested but I wasn't able to find anything. Since I want to merge this PR before the code freeze, please can you have another look ? 👀 

One more point, the behaviour of `translateEscape()` API was not correct, so I remove every escaping function in the method

File: engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/ActivityTest.java [hunk 1]
```diff
@@ -78,7 +78,7 @@ public final class ActivityTest {
                       .zeebeInput("https://github.com/{{orgId}}/{{repoId}}", "urlStatic")
                       .zeebeInputExpression(
                           "\"https://github.com/{{orgId}}/{{repoId}}\"", "urlExpression")
-                      .zeebeInput("My Name is \"Zeebe\", nice to meet you", "quotesStatic")
+                      .zeebeInput("My Name is \\\"Zeebe\\\", nice to meet you", "quotesStatic")
```
Comment (saig0) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/ActivityTest.java:81]: :x: This is different from the reported issue.

In the issue, the static input was `Hello\n\nYOU!`. With this change, we escape the double quotes. I don't expect a user or the modeler to write an escaped quote. 

In the Modeler, the double quotes are encoded as `&#34;`. So, the example would be `"My Name is &#34;Zeebe&#34;, nice to meet you"`.

I suggest aligning with the modeler.

File: engine/src/test/java/io/camunda/zeebe/engine/processing/variable/mapping/VariableMappingTransformerTest.java [hunk 2]
```diff
@@ -97,6 +97,26 @@ public void shouldEvaluateWithNotExistingVariable() {
         .isEqualTo(ResultType.OBJECT);
   }
 
+  @Test
+  public void shouldNotEscapeCharactersInStaticExpression() {
```
Comment (saig0) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/variable/mapping/VariableMappingTransformerTest.java:101]: :wrench: Let's add a test case for the double quotes encoded as `&#34;`.

File: engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/VariableMappingTransformer.java [hunk 3]
```diff
@@ -155,9 +154,7 @@ public String onEntry(final String targetKey, final Expression sourceExpression)
         final String expression;
 
         if (sourceExpression instanceof StaticExpression) {
-          expression =
-              String.format(
-                  "\"%s\"", StringEscapeUtils.escapeJava(sourceExpression.getExpression()));
+          expression = String.format("\"%s\"", sourceExpression.getExpression());
```
Comment (saig0) [inline engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/transformer/VariableMappingTransformer.java:157]: No escaping. No problem. :laughing:

**Review by saig0 — CHANGES_REQUESTED** [review]
@nicpuppa good catch. :+1: 

I would like to request a change for the test cases. Please have a look. :cake:

**Comment by nicpuppa** [comment]
> @nicpuppa BTW: I found out that the static inputs are used from the connector templates. 🤯 For example, the REST connector:
> 
> Screenshot from 2023-12-19 14-19-38

Yeah, but this should be covered by the test cases that we have right ? They are using static inputs inside the service task

**Comment by saig0** [comment]
> Yeah, but this should be covered by the test cases that we have right ?

Yes. This case is covered by `ActivityTest`. :+1: 

I was just not aware that it is used by the connector templates. :smile:

**Comment by nicpuppa** [comment]
> I was just not aware that it is used by the connector templates. 😄

Oh ok, now we are aware 😄 Good discover 🚀

**Comment by nicpuppa** [comment]
@saig0 I updated the test cases as suggested, please have another look 🙇

**Review by saig0 — APPROVED** [review]
@nicpuppa nice. :rocket: 

bors merge

**Pull request merged**